### PR TITLE
add AYON context to the AYON menu

### LINKED ISSUE #11: Show AYON context in AYON Menu
### Is there an existing issue for this?

- [X] I have searched the existing issues.

### Please describe the feature you have in mind and explain what the current shortcomings are?

Other AYON Addons include AYON Context on the top of the AYON menu.
image

### How would you imagine the implementation of the feature?

.

### Are there any labels you wish to add?

- [X] I have added the relevant labels to the enhancement request.

### Describe alternatives you've considered:

_No response_

### Additional context:

_No response_

### PR DESCRIPTION
**Pull request by MustafaJafar**
## Changelog Description
resolve #11 

Animation_47

## Testing notes:

1. Launch c4d via AYON Launcher
2. Check if AYON context is shown properly.

### CODE REVIEW

**Comment by BigRoy** [comment]
I'm quite confident that this will not update the label on folder/task changes - it's not live but the menu is built one. We'd need a dynamic menu label otherwise (not sure how to do that).

Which means the label that would be displayed would not be accurate.

---

Here's a younger me on how to define a custom menu for Cinema4D: https://developers.maxon.net/forum/topic/12103/custom-menu-in-c4d-top-menu-bar-with-custom-python-commands

This http://developers.maxon.net/forum//post/70228 seems to describe there is _some_ way to update menus later. But I'm not sure how/where exactly.

**Comment by MustafaJafar** [comment]
I was able to test this PR. 
The context label doesn't even appear. 
image

**Comment by MustafaJafar** [comment]
Now, the AYON context is shown BUT as @BigRoy mentioned above, it's static and not updated on context change.

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp [hunk 1]
```diff
@@ -60,6 +64,8 @@ AYON_RESET_RESOLUTION_ID = 1064318
 AYON_RESET_COLORSPACE_ID = 1064320
 AYON_EXPERIMENTAL_TOOLS_ID = 1064319
 
+AYON_CONTEXT_LABEL = 1064309
```
Comment (BigRoy) [inline client/ayon_cinema4d/startup/ayon_cinema4d.pyp:67]: What did you base this number on @MustafaJafar - is this officially registered with Maxon?

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp [hunk 2]
```diff
@@ -198,6 +204,11 @@ class ExperimentalTools(c4d.plugins.CommandData):
         return True
 
 
+class CONTEXTLABEL(c4d.plugins.CommandData):
```
Comment (BigRoy) [inline client/ayon_cinema4d/startup/ayon_cinema4d.pyp:207]: Why is this an all caps class name?

```suggestion
class ContextLabel(c4d.plugins.CommandData):
    """Stub menu entry to act as current context label"""
```

**Review by BigRoy — COMMENTED** [review]
Some minor notes - other than that this looks good!

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp (hunk 1 above)
Comment (MustafaJafar) [reply client/ayon_cinema4d/startup/ayon_cinema4d.pyp:67]: Honestly, I don't know. 

I think these are unique numbers to register the plugins/commands. I picked the smallest number - 1 in the pyp file.

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp (hunk 2 above)
Comment (MustafaJafar) [reply client/ayon_cinema4d/startup/ayon_cinema4d.pyp:207]: Lazy copying and pasting. Will fix it.

Comment (MustafaJafar) [reply client/ayon_cinema4d/startup/ayon_cinema4d.pyp]: https://github.com/ynput/ayon-cinema4d/pull/12/commits/572ff86130aa40fdf5df9dd80d4047062d16dfc1

**Comment by MustafaJafar** [comment]
In my last commit, I've added AYON icon. Please let me know if you don't like it :) @BigRoy 
image

File: client/ayon_cinema4d/api/pipeline.py [hunk 3]
```diff
@@ -263,3 +271,14 @@ def imprint_container(
     }
 
     lib.imprint(container, data, group="AYON")
+
+
+def on_task_changed():
+
+    if not is_headless_mode_enabled():
+        # Get AYON Context manu command plugin (menu item) by its unique id.
```
Comment (BigRoy) [inline client/ayon_cinema4d/api/pipeline.py:279]:
```suggestion
        # Get AYON Context menu command plugin (menu item) by its unique id.
```

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp (hunk 1 above)
Comment (BigRoy) [inline client/ayon_cinema4d/startup/ayon_cinema4d.pyp:67]: I registered this plugin id with maxon now.
```suggestion
AYON_CONTEXT_LABEL = 1064692
```

File: client/ayon_cinema4d/api/pipeline.py [hunk 4]
```diff
@@ -263,3 +271,14 @@ def imprint_container(
     }
 
     lib.imprint(container, data, group="AYON")
+
+
+def on_task_changed():
+
+    if not is_headless_mode_enabled():
+        # Get AYON Context menu command plugin (menu item) by its unique id.
+        ayon_context = c4d.plugins.FindPlugin(1064309)
```
Comment (BigRoy) [inline client/ayon_cinema4d/api/pipeline.py:280]: Update to registered plugin id
```suggestion
        ayon_context = c4d.plugins.FindPlugin(1064692)
```

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp [hunk 5]
```diff
@@ -60,6 +64,8 @@ AYON_RESET_RESOLUTION_ID = 1064318
 AYON_RESET_COLORSPACE_ID = 1064320
 AYON_EXPERIMENTAL_TOOLS_ID = 1064319
 
+AYON_CONTEXT_LABEL = 1064692
```
Comment (BigRoy) [inline client/ayon_cinema4d/startup/ayon_cinema4d.pyp:67]:
```suggestion
AYON_CONTEXT_LABEL_ID = 1064692
```

File: client/ayon_cinema4d/startup/ayon_cinema4d.pyp [hunk 6]
```diff
@@ -198,8 +210,14 @@ class ExperimentalTools(c4d.plugins.CommandData):
         return True
 
 
+class ContextLabel(c4d.plugins.CommandData):
+    id = AYON_CONTEXT_LABEL
```
Comment (BigRoy) [inline client/ayon_cinema4d/startup/ayon_cinema4d.pyp:214]:
```suggestion
    id = AYON_CONTEXT_LABEL_ID
```

**Review by BigRoy — APPROVED** [review]
Thanks so much - looking very nice!

**Pull request merged**